### ISSUE TITLE
cannot remove extrinsic from delayed_extrinsics queue for some special condition

### ISSUE THREAD

**Issue by xiaoch05** [darwinia-network/bridger#160]
https://github.com/darwinia-network/bridger/blob/20a5a3acdbc71845be8a8b425ac0d55c865c4a9b/src/service/subscribe/mod.rs#L144

if `need_to_sign_mmr_root_of` returns false, the `self.delayed_extrinsics` will never be removed

**Comment by HackFisher** [comment]
@xiaoch05 Please help check wether this issue still exist in latest master branch

**Comment by HackFisher** [comment]
```
/// A New MMR Root Scheduled Request to be Signed. [block number of the mmr root to sign]
		ScheduleMMRRoot(BlockNumber),
```

delayed_extrinsics is only used for `ScheduleMMRRoot` for now, can be deleted after we fix #303

**Comment by fewensa** [comment]
Close it, follow #303
